src/components/SelectorOptions/SelectorOptions.js

```diff
@@ -0,0 +1,476 @@
+import ContainerWithHeader from 'components/Layout/ContainerWithHeader';
+import Tabs from 'components/Tabs';
+
+import { fontSizes, spacing, fontStyles } from 'utils/variables';
+import { getThemeColors, themedColors } from 'utils/themes';
+import { images } from 'utils/images';
+import { getMatchingSortedData } from 'utils/textInput';
+import { isValidAddress } from 'utils/validators';
+
+import type { Theme } from 'models/Theme';
+import type { HorizontalOption, Option, OptionTabs } from 'models/Selector';
+
+
+type Props = {
+  horizontalOptionsData?: HorizontalOption[],
+  showOptionsTitles?: boolean,
+  renderOption?: (option: Option, onSelect: (option: Option) => void) => void,
+  onHide?: () => void,
+  onOptionSelect?: (option: Option, onSuccess: () => void) => void,
+  optionKeyExtractor?: (item: Object) => string,
+  isVisible: boolean,
+  title: string,
+  options?: Option[],
+  optionTabs?: OptionTabs[],
+  optionsTitle?: string,
+  searchPlaceholder?: string,
+  theme: Theme,
+  noImageFallback?: boolean,
+  inputIconName?: string,
+  iconProps?: Object,
+  activeAccountAddress: string,
+  onHidden: () => void,
+  validator?: (value: string) => string,
+  allowEnteringCustomAddress?: boolean,
+  forceTab?: string,
+};
+
+type State = {
+  query: string,
+  hasSearchError: boolean,
+  customAddressAsAnOption: ?Option,
+  activeTab?: string,
+};
+
+
+const DIAMETER = 64;
+
+const OptionsHeader = styled(MediumText)`
+  margin: ${spacing.large}px ${spacing.layoutSides}px 0;
+  ${fontStyles.regular};
+  color: ${themedColors.secondaryText};
+`;
+
+const HorizontalOptions = styled.View`
+  border-bottom-width: 1px;
+  border-style: solid;
+  border-color: ${themedColors.border};
+  padding-bottom: ${spacing.small}px;
+`;
+
+const HorizontalOptionItem = styled.TouchableOpacity`
+  align-items: center;
+  width: 90px;
+  padding-top: ${spacing.medium}px;
+`;
+
+const HorizontalOptionItemName = styled(BaseText)`
+  ${fontStyles.small};
+  color: ${themedColors.secondaryText};
+  padding: 0 4px;
+  margin-top: 8px;
+`;
+
+const EmptyStateWrapper = styled.View`
+  padding-top: 90px;
+  padding-bottom: 90px;
+  align-items: center;
+`;
+
+const SearchBarWrapper = styled.View`
+  padding: ${spacing.mediumLarge}px ${spacing.layoutSides}px 0;
+`;
+
+const IconCircle = styled.View`
+  width: ${DIAMETER}px;
+  height: ${DIAMETER}px;
+  border-radius: ${DIAMETER / 2}px;
+  background-color: ${themedColors.tertiary};
+  align-items: center;
+  justify-content: center;
+  text-align: center;
+  overflow: hidden;
+`;
+
+const IconImage = styled(CachedImage)`
+  height: ${DIAMETER}px;
+  width: ${DIAMETER}px;
+`;
+
+
+const viewConfig = {
+  minimumViewTime: 300,
+  viewAreaCoveragePercentThreshold: 100,
+  waitForInteraction: true,
+};
+
+const MIN_QUERY_LENGTH = 2;
+
+
+class SelectorOptions extends React.Component<Props, State> {
+  searchInput: TextInput;
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      query: '',
```

Comment:
I know `''` is falsey but i always prefer `null`... just a personal preference. I always worry that some tools or code evaluate `''` to be _something_ as opposed to _nothing_.